### PR TITLE
Config to fetch compare over time data

### PR DESCRIPTION
**Pull request by Carla-Moz**
In this PR, I'll breakdown my commits to add the functionality for fetching the data for the compare over time component. Although it's WIP, please feel free to review and comment as needed. 

closes issue https://mozilla-hub.atlassian.net/browse/PCF-383

- [x] [add the compare over time fetch logic to treeherder.ts ](https://github.com/mozilla/perfcompare/pull/650/commits/2b89ac6486d6ef05e1f01a76b898d8aac6f12097)
- [ ] [add logic to loader.ts to get compare over time results and interval values ](https://github.com/mozilla/perfcompare/pull/650/commits/2b58bc31fc0243a8ba3f6598f563f7a50c182e42) [obsolete - redone from comments. ]
- [x] [pass down the interval information to `Compare over Time` comp ](https://github.com/mozilla/perfcompare/pull/650/commits/3aeb3307b92de37cf2055cae14de869e0e2fa53b)
- [x] [created a separate loader and results view filed for Compare Over Time results ](https://github.com/mozilla/perfcompare/pull/650/commits/2c3654340a3be187cf468fd63fa2b967e0f75cc2)
- [x] [added a new route for compare over time results page ](https://github.com/mozilla/perfcompare/pull/650/commits/2c3654340a3be187cf468fd63fa2b967e0f75cc2)
- [ ] add test coverage 

### Concern: 
I've profiled my patch (yay profiler ! - honestly learned so much profiling a gazillion times lol) and the jank is pretty significant for loading the results page for compare over time. It looks like the biggest problem is with the rendering, specifically React's `performWorkUntilDeadline` method. If we could brainstorm how to fix this, I'd appreciate it to move this patch along faster.

### CODE REVIEW

File: src/components/CompareResults/loader.ts [hunk 1]
```diff
@@ -66,6 +55,67 @@ function checkValues({
...
+
+  if (newRevs.length !== newRepos.length) {
+    throw new Error(
+      'There should be as many "newRepo" parameters as there are "newRev" parameters.',
+    );
+  }
+
+  //if interval is not null, return values for time range component
+  if (interval !== null) {
+    if (interval === null) {
```
Comment (Carla-Moz) [inline src/components/CompareResults/loader.ts]: ignore this line; I removed it in commit #4 since of course it doesn't make sense :P

**Review by julienw — COMMENTED** [review]
Thanks for this work!

TLDR: it will be easier to write a new component instead of reusing ResultsView (but of course we'll reuse the ResultsMain), and a new loader file (but of course you can copy paste the existing one as a start).

Left a few comments, tell me what you think!

File: src/components/CompareResults/ResultsView.tsx [hunk 2]
```diff
@@ -50,6 +52,17 @@ function ResultsView(props: ResultsViewProps) {
           isBaseSearch={null}
           expandBaseComponent={() => null}
         />
+
+        {intervalValue ?? (
+          <CompareOverTime
```
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply src/components/CompareResults/ResultsView.tsx:57]: Yes, that makes sense! It got pretty knarly trying to fit CompareOverTime into the ResultsView with CompareWithBase. This will make it a lot simpler.

File: src/components/CompareResults/loader.ts
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply src/components/CompareResults/loader.ts:1]: Yes, this makes sense too. Again, I was trying to force fit OverTime workflow with WithBase and it was messy. I'll create a new loader file.

File: src/components/CompareResults/ResultsTable.tsx [hunk 3]
```diff
@@ -67,10 +68,13 @@ const customStyles = {
   background: 'none',
 };
 
-function ResultsTable() {
+function ResultsTable(props: { isOverTime: boolean }) {
   const themeMode = useAppSelector((state) => state.theme.mode);
+  const { isOverTime } = props;
+  const { results: withBaseResults } = useLoaderData() as LoaderReturnValue;
+  const { results: overTimeResults } = useLoaderData() as OverTimeLoadResults;
```
Comment (julienw) [inline src/components/CompareResults/ResultsTable.tsx:75]: My understanding is that the only reason you need the `isOvertime` prop is because of the different types. But the `results` property should be the same type, right? So maybe we can do something like:
```
  const { results } = useLoaderData() as { results: LoaderReturnValue['results'] };
```
Typescript doesn't care because for Typescript `useLoaderData()` is not really typed properly, so we can do whatever we want, as long as we do not lie.

I believe that with this change you can remove this new prop and simplify a bunch of code.

**Review by julienw — APPROVED** [review]
This looks pretty good to me, it's a lot simpler than the previous iteration!
I believe we can remove this prop `isOverTime`, see the comment below. Then you can land it without an additional review!

**Comment by julienw** [comment]
BTW Typescript doesn't pass, but this is because of `isOverTime` so this should be resolved once you remove it.
